google-cloud-firestore/lib/google/cloud/firestore/batch.rb

```diff
@@ -0,0 +1,1013 @@
+# Copyright 2017, Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/firestore/document"
+require "google/cloud/firestore/collection"
+require "google/cloud/firestore/query"
+require "google/cloud/firestore/convert"
+
+module Google
+  module Cloud
+    module Firestore
+      ##
+      # # Batch
+      #
+      # A batch in Cloud Firestore is a set of writes that execute
+      # atomically at a single logical point in time in a database.
+      #
+      # All changes are accumulated in memory until the block passed to
+      # {Database#batch} completes. Unlike transactions, batches are not
+      # automatically retried. See {Database#batch}.
```

Comment:
Why aren't they retried? 

This might help:

> Batched writes have fewer failure cases than transactions and use simpler code. They are not affected by contention issues, because they don't depend on consistently reading any documents.

Also adding a link to https://firebase.google.com/docs/firestore/manage-data/transactions is probably helpful.

Ditto for `Database#batch`.

Reply:
Added.

Reply:
~~ Accidentally removed @blowmage's comment on that he added it. ~~